cli/src/main/scala/com/codacy/analysis/cli/formatter/Formatter.scala

```diff
@@ -19,7 +20,9 @@ trait Formatter {
 
   def add(element: Result): Unit
 
-  def addAll(elements: Seq[Result]): Unit = elements.foreach(add)
+  def addAll(toolSpecification: Option[com.codacy.plugins.api.results.Tool.Specification],
```

Comment:
Adding more metadata about the tools to allow custom output with enriched information.